clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -172,10 +223,14 @@ private NetworkClientDelegate.UnsentRequest makeHeartbeatRequest() {
 
         membershipManager.groupInstanceId().ifPresent(data::setInstanceId);
 
-        if (this.subscriptions.hasPatternSubscription()) {
-            // TODO: Pass the string to the GC if server side regex is used.
-        } else {
-            data.setSubscribedTopicNames(new ArrayList<>(this.subscriptions.subscription()));
+        // Send subscription to the broker only if it has changed
+        if (sendUpdatedSubscription) {
```

Comment:
Makes sense, I will include the changes for the stateful builder with the initial approach, and that should be a solid ground to tune it afterwards and send only what's needed

Reply:
Done, I updated it back to sending all fields for now. I will follow up in a next PR to send only what's needed. 

I expect that it will be the existing `HeartbeatState` the one to extend, to be able to build a `ConsumerGroupRequestData` based on the last one sent, the member info, and the subscription info (determine difference to send only what changed, and send all on the failed attempts that it already handles for retry/backoff)